fix logic & add order module

jeagerism opened the pull request:
## 🛠️ What’s Changed

### ✅ **Refactoring & Improvements**

* **Refactored `usecase` initialization**:
  Removed manual injection of dependencies from `main.go` — usecases now initialize their own dependencies internally for better modularity.

* **Separated authentication logic:**

  * Split `Auth` and `Authorized` into distinct responsibilities.
  * Introduced `UserAuth` and `ShopAuth` for more granular access control.

* **Updated route access control:**

  * Changed `Login` and `Register` routes from `public` to `ignore` to make them explicitly bypass authorization, improving clarity.

* **Sensitive data handling:**

  * Removed `email` from JWT claims to reduce exposure of user data in payloads and prevent interception vulnerabilities.

---

### 🆕 New Feature: Order Module

Added basic order management functionality:

* `GetOrderByID`
* `GetOrderByShopID`
* `CreateOrder`

jeagerism committed:
getorderByID & getOrders for shop

jeagerism committed:
add user update status

tananut-pdkm commented on entity/order.go:
```diff
@@ -0,0 +1,109 @@
+	ShopID    uint `json:"shop_id"`
+	AddressID uint `json:"address_id"`
+	CourierID uint `json:"courier_id"`
+	Items     []struct {
```
We should separate this to the new struct because it's hard to reuse in the future

jeagerism committed:
shop change order status

jeagerism commented:
### 📌 Pull Request Description (shop chage order status commit)

#### 🛠 Changes (Refactor)

* Separated reusable `struct`s related to orders and statuses for future extensibility and maintainability.
* Moved logic from `delivery/shop.go` and `use.go` to:

  * `delivery/shop/http.go`
  * `delivery/user/http.go`
    to improve scalability and prepare for future use cases such as integrating with a message queue system (e.g., Kafka, RabbitMQ).

#### ✨ Additions (Feature)

* Added a new API: `UpdateOrderStatusByShopHandler`
  Allows a shop to update the status of an order with the following conditions:

  * **Pending** → can change to `Confirmed` or `Cancelled`
  * **Confirmed** → can change to `Shipped` or `Cancelled`
  * **Shipped** → can change to `Delivered` or `Cancelled`
  * **Delivered / Cancelled** → status cannot be changed further

jeagerism committed:
"test deploy"

jeagerism committed:
"go version"

jeagerism committed:
cloudbuild

jeagerism committed:
add cloudbuild.yaml

jeagerism committed:
cloudbuild

jeagerism committed:
change region

jeagerism committed:
change region

jeagerism committed:
add log

jeagerism committed:
log

jeagerism committed:
path

jeagerism committed:
fix

jeagerism committed:
final stage

jeagerism committed:
fix port

tananut-pdkm commented:
LGTM

jeagerism committed:
dockerfile

jeagerism committed:
dockerfile

jeagerism committed:
.

jeagerism committed:
rm docker-compose

jeagerism committed:
fix port

jeagerism committed:
change dockerfile

jeagerism committed:
add log

jeagerism committed:
versions

jeagerism committed:
test

jeagerism committed:
log

jeagerism committed:
test

jeagerism committed:
gemini

jeagerism committed:
a

jeagerism committed:
b

jeagerism committed:
c

jeagerism committed:
v

jeagerism committed:
2step

jeagerism committed:
d

jeagerism committed:
test1

jeagerism committed:
success
